internal/elasticsearch/ingest/datastream.go

```diff
@@ -75,12 +92,43 @@ func loadIngestPipelineFiles(dataStreamPath string, nonce int64) ([]Pipeline, er
 			Path:    path,
 			Name:    getPipelineNameWithNonce(name[:strings.Index(name, ".")], nonce),
 			Format:  filepath.Ext(path)[1:],
-			Content: c,
+			Content: append(c, routingPipeline...),
```

Comment:
I think we should add the `reroute` processor to the default pipeline only.

For example, here are the pipeline source and what Fleet generated during the package install:

Source:

```yaml
description: Pipeline for Kubernetes container logs
processors:
  - set:
      field: service.name
      copy_from: kubernetes.labels.app_kubernetes_io/name
      ignore_empty_value: true
on_failure:
  - set:
      field: event.kind
      value: pipeline_error
  - append:
      field: error.message
      value: '{{{ _ingest.on_failure_message }}}'

```

Generated:

```json
{
  "logs-kubernetes.container_logs-1.46.0": {
    "description": "Pipeline for Kubernetes container logs",
    "processors": [
      {
        "set": {
          "field": "service.name",
          "copy_from": "kubernetes.labels.app_kubernetes_io/name",
          "ignore_empty_value": true
        }
      },
      {
        "pipeline": {
          "name": "logs-kubernetes.container_logs@custom",
          "ignore_missing_pipeline": true
        }
      },
      {
        "reroute": {
          "tag": "kubernetes.container_logs",
          "dataset": [
            "{{kubernetes.labels.elastic_co/dataset}}",
            "{{data_stream.dataset}}",
            "kubernetes.container_logs"
          ],
          "namespace": [
            "{{kubernetes.labels.elastic_co/namespace}}",
            "{{data_stream.namespace}}",
            "default"
          ],
          "if": "ctx?.kubernetes?.labels != null"
        }
      }
    ],
    "on_failure": [
      {
        "set": {
          "field": "event.kind",
          "value": "pipeline_error"
        }
      },
      {
        "append": {
          "field": "error.message",
          "value": "{{{ _ingest.on_failure_message }}}"
        }
      }
    ],
    "_meta": {
      "managed_by": "fleet",
      "managed": true,
      "package": {
        "name": "kubernetes"
      }
    }
  }
}
```

Reply:
I ran a quick test, and Fleet added the reroute processor at the end of the processor list for the default pipeline only.

Reply:
Thanks @zmoog ! Does that mean if we have both `default.yml` and `third-party.yml` (for example) and `routing_rules.yml`, the reroute processors will only be attached to default.yml processors not the `third-party.yml` ?

Reply:
Yep I just tested with `third-party.yml` in Kibana and `routing_rules.yml` only got attached to the `default.yml` pipeline. Will make sure this is the behavior with this PR! Thank you!

Reply:
Yeah, exactly! Kibana only injects the rules into the default pipeline (defined in the `default.yml` file). 

I'm sorry, but somehow I missed your comment from Aug 1.

Reply:
Done :) Good catch! Thanks!